keepsorted/options_parser.go

```diff
@@ -76,11 +85,82 @@ func (p *parser) popInt() (int, error) {
 }
 
 func (p *parser) popList() ([]string, error) {
+	if p.allowYAMLLists {
+		val, rest, ok := tryFindYAMLListAtStart(p.line)
+		if ok {
+			p.line = rest
+			return parseYAMLList(val)
+		}
+	}
 	val, rest, _ := strings.Cut(p.line, " ")
 	p.line = rest
 	return strings.Split(val, ","), nil
 }
 
+func tryFindYAMLListAtStart(s string) (list, rest string, ok bool) {
+	if s == "" || s[0] != '[' {
+		return "", "", false
+	}
+
+	var quote rune
+	var depth int // s[0] == '[' forces this to 1 after the first iteration.
+	iter := newRuneIter(s)
+	for {
+		ch, ok := iter.pop()
+		if !ok {
+			break
+		}
+		switch ch {
+		case '[':
+			if quote == 0 {
+				depth++
+			}
+		case ']':
+			if quote == 0 {
+				if depth > 1 {
+					depth--
+					continue
+				}
+
+				// depth == 1
+
+				// Force the last ] to either be the end, or followed by a space
+				// We don't want to allow
+				// key1=[a, b, c ,d]key2=yes
+				if next, ok := iter.peek(); !ok || next == ' ' {
+					depth--
+				}
+				break
```

Comment:
This break feels like the intent was to break out of the for loop. After all, this is when we found the end of the list and we want to stop processing. But we're in a `switch`, so doesn't `break` actually break out of the `switch` and the loop keeps going?

It seems to be a testing gap, too, as no test cases appear to have parameters after the YAML list.